salt/orchestrate/edx/init.sls

```diff
@@ -72,24 +71,34 @@ populate_mine_with_edx_node_data:
 reload_pillar_data_on_edx_nodes:
   salt.function:
     - name: saltutil.refresh_pillar
-    - tgt: 'G@roles:edx and G@environment:{{ ENVIRONMENT }}'
+    - tgt: 'P@roles:(edx|edx-worker) and G@environment:{{ ENVIRONMENT }}'
     - tgt_type: compound
     - require:
         - salt: populate_mine_with_edx_node_data
 
 {# Deploy Consul agent first so that the edx deployment can use provided DNS endpoints #}
 deploy_consul_agent_to_edx_nodes:
   salt.state:
-    - tgt: 'G@roles:edx and G@environment:{{ ENVIRONMENT }}'
+    - tgt: 'P@roles:(edx|edx-worker) and G@environment:{{ ENVIRONMENT }}'
     - tgt_type: compound
     - sls:
         - consul
         - consul.dns_proxy
 
 build_edx_nodes:
   salt.state:
-    - tgt: 'G@roles:edx and G@environment:{{ ENVIRONMENT }}'
+    - tgt: 'P@roles:(edx|edx-worker) and G@environment:{{ ENVIRONMENT }}'
     - tgt_type: compound
     - highstate: True
     - require:
         - salt: deploy_consul_agent_to_edx_nodes
+
+stop_non_edx_worker_services:
+{% for service in ['edxapp:', 'forum', 'xqueue', 'xqueue_consumer'] %}
```

Comment:
The for loop needs to be above the state ID and the ID needs to be templatized so that it doesn't violate the uniqueness constraint.